src/common/meta/src/reconciliation/reconcile_table/reconcile_regions.rs

```diff
@@ -0,0 +1,199 @@
+// limitations under the License.
+
+use std::any::Any;
+use std::collections::{HashMap, HashSet};
+
+use api::v1::column_def::try_as_column_def;
+use api::v1::region::region_request::Body;
+use api::v1::region::{
+    alter_request, AlterRequest, RegionColumnDef, RegionRequest, RegionRequestHeader, SyncColumns,
+};
+use api::v1::{ColumnDef, SemanticType};
+use async_trait::async_trait;
+use common_procedure::{Context as ProcedureContext, Status};
+use common_telemetry::info;
+use common_telemetry::tracing_context::TracingContext;
+use futures::future;
+use serde::{Deserialize, Serialize};
+use snafu::{OptionExt, ResultExt};
+use store_api::metadata::ColumnMetadata;
+use store_api::metric_engine_consts::TABLE_COLUMN_METADATA_EXTENSION_KEY;
+use store_api::storage::{ColumnId, RegionId};
+
+use crate::ddl::utils::{add_peer_context_if_needed, extract_column_metadatas};
+use crate::error::{ConvertColumnDefSnafu, Result, UnexpectedSnafu};
+use crate::reconciliation::reconcile_table::reconciliation_end::ReconciliationEnd;
+use crate::reconciliation::reconcile_table::update_table_info::UpdateTableInfo;
+use crate::reconciliation::reconcile_table::{ReconcileTableContext, State};
+use crate::rpc::router::{find_leaders, region_distribution};
+
+#[derive(Debug, Serialize, Deserialize)]
+pub struct ReconcileRegions {
+    column_metadatas: Vec<ColumnMetadata>,
+    region_ids: HashSet<RegionId>,
+}
+
+impl ReconcileRegions {
+    pub fn new(column_metadatas: Vec<ColumnMetadata>, region_ids: Vec<RegionId>) -> Self {
+        Self {
+            column_metadatas,
+            region_ids: region_ids.into_iter().collect(),
+        }
+    }
+}
+
+#[async_trait]
+#[typetag::serde]
+impl State for ReconcileRegions {
+    async fn next(
+        &mut self,
+        ctx: &mut ReconcileTableContext,
+        _procedure_ctx: &ProcedureContext,
+    ) -> Result<(Box<dyn State>, Status)> {
+        let table_meta = ctx.build_table_meta(&self.column_metadatas)?;
+        ctx.volatile_ctx.table_meta = Some(table_meta);
+        let table_id = ctx.table_id();
+        let table_name = ctx.table_name();
+
+        let primary_keys = self
+            .column_metadatas
+            .iter()
+            .filter(|c| c.semantic_type == SemanticType::Tag)
+            .map(|c| c.column_schema.name.to_string())
+            .collect::<HashSet<_>>();
+        let column_defs = self
+            .column_metadatas
+            .iter()
+            .map(|c| {
+                let column_def = try_as_column_def(
+                    &c.column_schema,
+                    primary_keys.contains(&c.column_schema.name),
+                )
+                .context(ConvertColumnDefSnafu {
+                    column: &c.column_schema.name,
+                })?;
+
+                Ok((c.column_id, column_def))
+            })
+            .collect::<Result<Vec<_>>>()?;
+
+        // Sends sync column metadatas to datanode.
+        // Safety: must exists.
```

Comment:
This "Safety" comment is useless. At least it should explain the "WHY" of must existence.